Amplify/Categories/Auth/Models/AuthCodeDeliveryDetails.swift

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct AuthCodeDeliveryDetails {
+
+    public let desitination: String?
+
+    public let deliveryMedium: DeliveryMedium
+
+    public let attribuetName: String?
+
+    public init(desitination: String?,
+                 deliveryMedium: DeliveryMedium,
+                 attribuetName: String?) {
```

Comment:
It seems we should have destination and attribute name be required right? I don't know if AWSMobileClient always returns them but it seems all three pieces of info are required for this object. So if it returns null we can for now fill them in with empty strings but from a framework standpoint we should require all based on the meaning I think.

Reply:
Agreed, made them non optional.